.github/workflows/test-all.yml

```
@@ -0,0 +1,190 @@
+          if [[ $EVENT_NAME == "schedule" ]]; then
+            echo "Setting release type to nightly"
+            echo "RELEASE_TYPE=nightly" >> $GITHUB_OUTPUT
+          elif [[ $EVENT_NAME == "push" && $REF == refs/tags/v* ]]; then
+            echo "Setting release type to release"
+            echo "RELEASE_TYPE=release" >> $GITHUB_OUTPUT
+          else
+            echo "Setting release type to dry-run"
+            echo "RELEASE_TYPE=dry-run" >> $GITHUB_OUTPUT
+          fi
+  prepare_hermes_workspace:
+    runs-on: ubuntu-latest
+    env:
+      HERMES_WS_DIR: /tmp/hermes
+      HERMES_VERSION_FILE: packages/react-native/sdks/.hermesversion
+      BUILD_FROM_SOURCE: true
+    outputs:
+      react-native-version: ${{ steps.react-native-version.outputs.version }}
+      hermes-version: ${{ steps.hermes-version.outputs.version }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+      - name: Setup node.js
+        uses: ./.github/actions/setup-node
+      - name: Setup hermes version
+        id: hermes-version
+        run: |
+          mkdir -p "/tmp/hermes" "/tmp/hermes/download" "/tmp/hermes/hermes"
+
+          if [ -f "$HERMES_VERSION_FILE" ]; then
+            echo "Hermes Version file found! Using this version for the build:"
+            echo "VERSION=$(cat $HERMES_VERSION_FILE)" >> "$GITHUB_OUTPUT"
+          else
+            echo "Hermes Version file not found!!!"
+            echo "Using the last commit from main for the build:"
+            HERMES_TAG_SHA=$(git ls-remote https://github.com/$GITHUB_REPOSITORY main | cut -f 1 | tr -d '[:space:]')
+            echo "VERSION=$HERMES_TAG_SHA" >> "$GITHUB_OUTPUT"
+          fi
+          echo "Hermes commit is $HERMES_TAG_SHA"
+      - name: Get react-native version
+        id: react-native-version
+        run: |
+          VERSION=$(cat packages/react-native/package.json | jq -r '.version')
+          # Save the react native version we are building in an output variable so we can use that file as part of the cache key.
+          echo "VERSION=$VERSION" >> "$GITHUB_OUTPUT"
+          echo "React Native Version is $VERSION"
+      - name: Cache hermes workspace
+        uses: actions/cache@v4.0.0
+        with:
+          path: |
+            /tmp/hermes/download/
+            /tmp/hermes/hermes/
+          key: v1-hermes-${{ steps.hermes-version.outputs.version }}-${{ github.run_number }}
+          enableCrossOsArchive: true
+      - name: Yarn- Install Dependencies
+        run: yarn install --non-interactive
+      - name: Download Hermes tarball
+        run: |
+          node packages/react-native/scripts/hermes/prepare-hermes-for-build ${{ github.event.pull_request.html_url }}
+          cp packages/react-native/sdks/download/* $HERMES_WS_DIR/download/.
+          cp -r packages/react-native/sdks/hermes/* $HERMES_WS_DIR/hermes/.
+
+          echo ${{ steps.hermes-version.outputs.version }}
+  build_andriod:
+    runs-on: ubuntu-latest
```

```suggestion
    runs-on: 8-core-ubuntu
```

Can we make this and other compilation heavy jobs on `8-core-ubuntu`? This should be similar to the same machine we had on CircleCI

---

You could, however these larger runners would need to be configured in your org with the label `8-core-ubuntu` in order to run. There are no runners with this label available by default.